main/src/cgeo/geocaching/sorting/SortActionProvider.java

```diff
@@ -171,4 +175,8 @@ public void setSelection(final CacheComparator selection) {
     public void setIsEventsOnly(final boolean isEventsOnly) {
         this.isEventsOnly = isEventsOnly;
     }
+
+    public void setDistanceToTargetAvailable(final boolean distanceToTargetAvailable) {
```

Comment:
Suggested solution: change this method as follows:
- Instead of passing a boolean, pass the actual target coordinate in this method and create a corresponding TargetDistanceComparator. 
- Store this as instance member instead of the boolean flag.
- replace checks for target availability with checks on whether the comparator is null
- use the comparator where needed